### PR TITLE
[Docs] Refactor config tutorials for 3.x.

### PR DESCRIPTION
**Pull request by RangiLyu**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMCls.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: docs/en/tutorials/config.md [hunk 1]
```diff
@@ -245,180 +166,245 @@ model = dict(
...
+
+```python
+_base_ = [
+    '../_base_/models/mask_rcnn_r50_fpn.py',
+    '../_base_/datasets/coco_instance.py',
+    '../_base_/schedules/schedule_1x.py', '../_base_/default_runtime.py'
+]
+```
+
+If you wish to inspect the config file, you may run `python tools/misc/print_config.py /PATH/TO/CONFIG` to see the complete config.
```
Comment (chhluo) [inline docs/en/tutorials/config.md:407]: image

need update

File: docs/en/tutorials/config.md [hunk 2]
```diff
@@ -542,3 +513,49 @@ model = dict(
...
+
+We follow the below style to name config files. Contributors are advised to follow the same style.
+
+```
+{model}_[model setting]_{backbone}_{neck}_[norm setting]_[misc]_[gpu x batch_per_gpu]_{schedule}_{dataset}
+```
+
+`{xxx}` is required field and `[yyy]` is optional.
+
+- `{model}`: model type like `faster_rcnn`, `mask_rcnn`, etc.
```
Comment (BIGWangYuDong) [inline docs/en/tutorials/config.md:548]: may update after refactoring the configs.

File: docs/en/tutorials/config.md [hunk 3]
```diff
@@ -245,180 +166,245 @@ model = dict(
...
+                   'scale_factor'))
 ]
-test_pipeline = [
-    dict(type='LoadImageFromFile'),  # First pipeline to load images from file path
+train_dataloader = dict(   # Train dataloader config
+    batch_size=2,  # Batch size of a single GPU
+    num_workers=2,  # Worker to pre-fetch data for each single GPU
+    persistent_workers=True,  # If ``True``, the dataloader will not shutdown the worker processes after an epoch end, which can accelerate training speed.
+    sampler=dict(  # training data sampler
+        type='DefaultSampler',  # DefaultSampler which supports both distributed and non-distributed training. Refer to https://github.com/open-mmlab/mmengine/blob/main/mmengine/dataset/sampler.py
```
Comment (BIGWangYuDong) [inline docs/en/tutorials/config.md:211]: I think it is necessary to give users an example of `iter-based` setting?

File: docs/en/tutorials/config.md [hunk 4]
```diff
@@ -245,180 +166,245 @@ model = dict(
...
+    batch_sampler=dict(type='AspectRatioBatchSampler'),  # batch sampler for grouping images with similar aspect ratio into a same batch. It can reduce GPU memory cost.
+    dataset=dict(  # Train dataset config
+        type=dataset_type,
+        data_root=data_root,
+        ann_file='annotations/instances_train2017.json',  # Path of annotation file
+        data_prefix=dict(img='train2017/'),  # Prefix of image path
+        filter_cfg=dict(filter_empty_gt=True, min_size=32),  # config of filtering images and annotations
+        pipeline=train_pipeline))
+val_dataloader = dict(  # Validation dataloader config
+    batch_size=1,  # Batch size of a single GPU
```
Comment (BIGWangYuDong) [inline docs/en/tutorials/config.md:222]: we can add comments to tell users setting bs=2 may degrade the performance

File: docs/en/tutorials/config.md (hunk 2 above)
Comment (RangiLyu) [reply docs/en/tutorials/config.md:586]: update later

File: docs/en/tutorials/config.md [hunk 5]
```diff
@@ -245,180 +166,283 @@ model = dict(
...
 dataset_type = 'CocoDataset'  # Dataset type, this will be used to define the dataset
 data_root = 'data/coco/'  # Root path of data
-img_norm_cfg = dict(  # Image normalization config to normalize the input images
-    mean=[123.675, 116.28, 103.53],  # Mean values used to pre-training the pre-trained backbone models
-    std=[58.395, 57.12, 57.375],  # Standard variance used to pre-training the pre-trained backbone models
-    to_rgb=True
-)  # The channel orders of image used to pre-training the pre-trained backbone models
-train_pipeline = [  # Training pipeline
-    dict(type='LoadImageFromFile'),  # First pipeline to load images from file path
+file_client_args = dict(backend='disk')  # file client arguments
```
Comment (BIGWangYuDong) [inline docs/en/tutorials/config.md:179]: `file client docs` for implementation details.

File: docs/en/tutorials/config.md [hunk 6]
```diff
@@ -245,180 +166,283 @@ model = dict(
...
 ]
-test_pipeline = [
-    dict(type='LoadImageFromFile'),  # First pipeline to load images from file path
+```
+
+### Hook config
+
+Users can attach hooks to training, validation, and testing loops to insert some oprations during running. There are two different hook fields, one is `default_hooks` and the other is `custom_hooks`.
+
+`default_hooks` is a dict of hook configs. `default_hooks` are the hooks must required at runtime. They have default priority which should not be modified. If not set, runner will use the default values. To disable a default hook, users can set its config to `None`.
```
Comment (BIGWangYuDong) [inline docs/en/tutorials/config.md:330]: we can give tutorial docs here

File: docs/en/tutorials/config.md [hunk 7]
```diff
@@ -245,180 +166,283 @@ model = dict(
...
+
+```python
+_base_ = [
+    '../_base_/models/mask_rcnn_r50_fpn.py',
+    '../_base_/datasets/coco_instance.py',
+    '../_base_/schedules/schedule_1x.py', '../_base_/default_runtime.py'
+]
+```
+
+If you wish to inspect the config file, you may run `python tools/misc/print_config.py /PATH/TO/CONFIG` to see the complete config.
```
Comment (BIGWangYuDong) [inline docs/en/tutorials/config.md:445]: I think we can add an example such as:
```
# 有时我们想重复利用 _base_ 中定义的字段内容，就可以通过 {{_base_.xxxx}} 获取来获取对应变量的拷贝。例如：

_base_ = ['resnet50.py']
a = {{_base_.model}}
# 等价于 a = dict(type='ResNet', depth=50)
```

File: docs/en/tutorials/config.md (hunk 6 above)
Comment (RangiLyu) [reply docs/en/tutorials/config.md:330]: I'll add a link to mmengine's tutorial

File: docs/en/tutorials/config.md [hunk 8]
```diff
@@ -245,180 +166,283 @@ model = dict(
...
+        filter_cfg=dict(filter_empty_gt=True, min_size=32),  # config of filtering images and annotations
+        pipeline=train_pipeline))
+val_dataloader = dict(  # Validation dataloader config
+    batch_size=1,  # Batch size of a single GPU. If batch-szie > 1, the extra padding area may influence the performance.
+    num_workers=2,  # Worker to pre-fetch data for each single GPU
+    persistent_workers=True,  # If ``True``, the dataloader will not shutdown the worker processes after an epoch end, which can accelerate training speed.
+    drop_last=False,  # Whether to drop the last incomplete batch, if the dataset size is not divisible by the batch size
+    sampler=dict(
+        type='DefaultSampler',
+        shuffle=False),  # not shuffle the during validation and testing
```
Comment (chhluo) [inline docs/en/tutorials/config.md:228]: not shuffle during validation and testing

File: docs/en/tutorials/config.md [hunk 9]
```diff
@@ -245,180 +166,283 @@ model = dict(
...
+        test_mode=True,  # Turn on test mode of the dataset to avoid filtering annotations or images
+        pipeline=test_pipeline))
+test_dataloader = val_dataloader  # Testing dataloader config
+```
+
+[Evaluators](https://mmengine.readthedocs.io/en/latest/tutorials/metric_and_evaluator.html) are used to compute the metrics of the trained model on the validation and testing datasets. The config of evaluators consists of one or a list of metric configs:
+
+```python
+val_evaluator = dict(  # Validation evaluator config
+    type='CocoMetric',  # The coco metric used to evaluate AR, AP, and mAP for detection and instance segmentation
```
Comment (chhluo) [inline docs/en/tutorials/config.md:243]: add the code link to `CocoMetric`

File: docs/en/tutorials/config.md [hunk 10]
```diff
@@ -478,7 +501,8 @@ model = dict(
                 num_branches=4,
                 block='BASIC',
                 num_blocks=(4, 4, 4, 4),
-                num_channels=(32, 64, 128, 256)))),
+                num_channels=(32, 64, 128, 256))),
+        init_cfg=dict(type='Pretrained', checkpoint='open-mmlab://msra/hrnetv2_w32')),
     neck=dict(...))
 ```
```
Comment (chhluo) [inline docs/en/tutorials/config.md:508]: How about the old keys which are different to new keys?

File: docs/en/tutorials/config.md [hunk 11]
```diff
@@ -542,3 +551,49 @@ model = dict(
...
+  e.g. `[dict(type='LoadImageFromFile'), ...]`. If you want to change `'LoadImageFromFile'` to `'LoadImageFromNDArray'` in the pipeline,
+  you may specify `--cfg-options data.train.pipeline.0.type=LoadImageFromNDArray`.
+
+- Update values of list/tuples.
+
+  If the value to be updated is a list or a tuple. For example, the config file normally sets `model.data_preprocessor.mean=[123.675, 116.28, 103.53]`. If you want to
+  change the mean values, you may specify `--cfg-options model.data_preprocessor.mean="[127,127,127]"`. Note that the quotation mark `"` is necessary to
+  support list/tuple data types, and that **NO** white space is allowed inside the quotation marks in the specified value.
+
+## Config name style
```
Comment (ZwwWayne) [inline docs/en/tutorials/config.md:576]: should update according to the new convention of OpenMMLab 2.0

File: docs/en/tutorials/config.md (hunk 5 above)
Comment (RangiLyu) [reply docs/en/tutorials/config.md:179]: MMEngine currently does not have file client docs. I'll update it later.

File: docs/en/tutorials/config.md (hunk 9 above)
Comment (RangiLyu) [reply docs/en/tutorials/config.md:243]: CocoMetric will be moved to mmeval later.

File: docs/en/tutorials/config.md [hunk 12]
```diff
@@ -1,106 +1,27 @@
...
-we make brief comments on the config of Mask R-CNN using ResNet50 and FPN as the following.
-For more detailed usage and the corresponding alternative for each modules, please refer to the API documentation.
+In mmdetection's config, we use `model` to setup detection algorithm components. In addition to neural network components such as `backbone`, `neck` etc, it also requires `data_preprocessor`, `train_cfg`, and `test_cfg`. `data_preprocessor` is responsible for processing a batch of data output by dataloader. `train_cfg`, and `test_cfg` in the model config are for training and testing hyperparameters of the components.
 
 ```python
 model = dict(
     type='MaskRCNN',  # The name of detector
+    data_preprocessor=dict(  # The config of data preprocessor, usually includes image normalization and padding
+        type='DetDataPreprocessor',  # The type of the data preprocessor, refer to https://mmdetection.readthedocs.io/en/dev-3.x/api.html#module-mmdet.models.data_preprocessors
+        mean=[123.675, 116.28, 103.53],  # Mean values used to pre-training the pre-trained backbone models
```
Comment (chhluo) [inline docs/en/tutorials/config.md:18]: add description about the channel order of pixel mean and std.

File: docs/en/tutorials/config.md [hunk 13]
```diff
@@ -491,54 +515,92 @@ It's worth noting that when modifying intermediate variables in the children con
...
+```
+
+The file name is divided to four parts. All parts and components are connected with `_` and words of each part or component should be connected with `-`.
+
+- `{algorithm name}`: The name of the algorithm. It can be a detector name such as `faster-rcnn`, `mask-rcnn`, etc. Or can be a semi-supervise or knowladge-distillation algorithm such as `soft-teacher`, `lad`. etc.
+- `{model component names}`: Names of the components used in the algorithm such as backbone, neck, etc. For example, `r50-caffe_fpn_gn-head` means using caffe-style ResNet50, FPN and detection head with Group Norm in the algorithm.
+- `{training settings}`: Information of training settings such as batch size, augmentations, loss trick, scheduler, and epochs/iterations. For example: `4xb4-mixup-giou-coslr-100e` means using 8-gpus x 4-images-per-gpu, mixup augmentation, GIoU loss, cosine annealing learning rate, and train 100 epochs.
+  Some abbreviations:
+  - `{gpu x batch_per_gpu}`: GPUs and samples per GPU. E.g. `4x4b` is the short term of 4-gpus x 4-images-per-gpu. And `8xb2` is used by default if not mentioned.
```
Comment (ZwwWayne) [inline docs/en/tutorials/config.md:600]: 4xb4, we can say bN indicates N batch size per GPU

**Review by ZwwWayne — APPROVED** [review]
